qiskit_ibm_runtime/runtime_job.py

```diff
@@ -262,14 +249,12 @@ def stream_results(
             RuntimeInvalidStateError: If a callback function is already streaming results or
                 if the job already finished.
         """
+        if self._status in JOB_FINAL_STATES:
```

Comment:
Not sure if/how this logic needs to be updated? If wait_for_final_state times out and stream_results is called, the error messages would be inaccurate

Reply:
also there are still a couple failing integration tests in push-test